src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -133,7 +133,7 @@ CHIP_ERROR UDPEndPointImplLwIP::LwIPBindInterface(struct udp_pcb * aUDP, Interfa
 InterfaceId UDPEndPointImplLwIP::GetBoundInterface() const
 {
 #if HAVE_LWIP_UDP_BIND_NETIF
-    return InterfaceId(netif_get_by_index(mUDP->netif_idx));
+    return InterfaceId(netif_get_by_index(++(mUDP->netif_idx)));
```

Comment:
Increase interface index by one each time seems wrong.
Also this seems to permanently increase things (how does this compile? Is mUDP not const?).

In any case, could it be that we use netif_idx being 0 as a whildcard?
In which case this would be:

```cpp
return (mUdp->netif_idx == 0)
    ? InterfaceId::Null()
    :  InterfaceId(netif_get_by_index(++(mUDP->netif_idx)));
```

Or something similar? I do not know if the above works or is correct, but just blindly increasing interface index seems wrong.

Reply:
Specifically it looks like `udp_new_ip_type` calls `udp_new` which does a `memset(pcb, 0, sizeof(struct udp_pcb));`.

This is why it looks like mUDP netif_idx starts off as 0. Need to read more documentation if 0 is correct or something else is going on.

Reply:
Within LWIP code it looks like checks like this exist:

```c
  /* check if PCB is bound to specific netif */
  if ((pcb->netif_idx != NETIF_NO_INDEX) &&
      (pcb->netif_idx != netif_get_index(ip_data.current_input_netif))) {
    return 0;
  }
```

lwip/netif.h defines `#define NETIF_NO_INDEX              (0)`.

So my current guess of correct code may be:

```cpp
return  (mUdp->netif_idx == NETIF_NO_INDEX) ? InterfaceId::Null() : InterfaceId(netif_get_by_index(mUDP->netif_idx));
```

Reply:
I am also unclear if the above code would fix anything ... netif_get_by_index returns NULL if index is NETIF_NO_INDEX at least in the copy I am looking at, so the previous code may be equivalent.

Reply:
I believe our current code does not handle null netif in joining the multicast group.

We only seem to use `mld6_joingroup_netif` which requires a real netif. Instead, I believe in `UDPEndPointImplLwIP::IPv6JoinLeaveMulticastGroupImpl` we should handle null interfaceid (check `isPresent`) and we should use `mld6_joingroup`/`mld6_leavegroup` if no specific interface is set.

Reply:
we should probably also fix IPv4 for completeness while we are at it.

Reply:
No need to test, but please fix to at least compile. .... I can do that change on your branch.